org.eclipse.winery.repository/src/main/java/org/eclipse/winery/repository/backend/BackendUtils.java

```diff
@@ -712,120 +712,116 @@ public static void persist(DefinitionsChildId id, Definitions definitions) throw
     public static void persist(Object o, RepositoryFileReference ref, MediaType mediaType) throws IOException {
         // We assume that the object is not too large
         // Otherwise, http://io-tools.googlecode.com/svn/www/easystream/apidocs/index.html should be used
-        ByteArrayOutputStream out = new ByteArrayOutputStream();
-        Marshaller m;
-        try {
-            m = JAXBSupport.createMarshaller(true);
+        try (ByteArrayOutputStream out = new ByteArrayOutputStream()) {
+            Marshaller m = JAXBSupport.createMarshaller(true);
             m.marshal(o, out);
+            byte[] data = out.toByteArray();
+            try (ByteArrayInputStream in = new ByteArrayInputStream(data)) {
+                // this may throw an IOException. We propagate this exception.
+                RepositoryFactory.getRepository().putContentToFile(ref, in, mediaType);
```

Comment:
this code (just like the previously existing code) works under the assumption that there is only ever one IRepository actively in use on a JVM. That assumption will not be true in the container (as soon as the refactor to use the winery model there is merged). As such this may write to a different repository than the ref belongs to.

Reply:
True. In the "container" case this could indeed be an issue. However, I think we should in general avoid using Winery's util classes whenever possible as they are developed for the use in Winery ... and not for a "library" use case as in the container.

Reply:
correct, but "public" API we use (like getting a Definition from a Repository) may have calls like this on their codepath, resulting in somewhat weirdly-behaving bugs.